libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -83,55 +90,59 @@ Compass::start_calibration_mask(uint8_t mask, bool retry, bool autosave, float d
 bool
 Compass::start_calibration_all(bool retry, bool autosave, float delay, bool autoreboot)
 {
+    _cal_autosave = autosave;
+    _compass_cal_autoreboot = autoreboot;
+
     for (uint8_t i=0; i<COMPASS_MAX_INSTANCES; i++) {
-        if (healthy(i) && use_for_yaw(i)) {
-            if (!start_calibration(i,retry,autosave,delay,autoreboot)) {
-                cancel_calibration_all();
-                return false;
-            }
-        }
+        // ignore any compasses that fail to start calibrating
+        // start all should only calibrate compasses that are being used
+        _start_calibration(i,retry,delay);
     }
     return true;
```

Comment:
Good point, forgot to change that aspect. I wonder if this should reject the start though to run the check for hal.util->get_soft_armed() here rather then in the mavlink parser. (Compass cal from copter will call the start calibration if motors aren't armed, but that ignores the soft armed state, I'm not clear on the differences between motor arming and soft armed but I wonder if we should be moving the protection)